conans/client/file_copier.py

```diff
@@ -237,6 +239,7 @@ def _copy_files(files, src, dst, keep_path, symlinks):
                     pass
                 os.symlink(linkto, abs_dst_name)  # @UndefinedVariable
             else:
-                shutil.copy2(abs_src_name, abs_dst_name)
+                if not if_different or not os.path.exists(abs_dst_name) or not filecmp.cmp(abs_src_name, abs_dst_name):
```

Comment:
If the reason is performance, some data/benchmark about the savings would be great.